### ISSUE TITLE
Mac: Handle failures writing and deleting placeholders when re-expanding folders as part of a projection change

### ISSUE THREAD

**Issue by wilbaker** [microsoft/VFSForGit#245]
Follow up work item for #239, see TODO's in code for specific spots that need to be addressed

**Comment by wilbaker** [comment]
Re-opening issue.  I'm not sure that the changes in #1014 addressed the TODO here:

https://github.com/microsoft/VFSForGit/blob/2df2b01c4cec71f83b5d049d0e20cc5e77b5b464/GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs#L1439

I'm thinking if the placeholder write failed we should add that path to ModifiedPaths.dat.

@kewillford what do you think?

**Comment by kewillford** [comment]
You are right the one that those changes fixed had to do with deleting placeholders that were already on disk not the initial writing of one.
